server/controller/src/main/java/ai/starwhale/mlops/domain/job/storage/JobRepo.java

```diff
@@ -0,0 +1,495 @@
+                    .rawResult(true)
+                    .build();
+            @Override
+            public boolean hasNext() {
+                return !finished;
+            }
+
+            @Override
+            public List<JobFlattenEntity> next() {
+                var records = store.query(request).getRecords();
+                if (records.isEmpty()) {
+                    finished = true;
+                    return List.of();
+                }
+                // update for next request
+                request.setStart(request.getStart() + records.size());
+
+                var entities = objectMapper.convertValue(records, new TypeReference<List<JobFlattenEntity>>() {
+                });
+                // populate bean
+                for (JobFlattenEntity job : entities) {
+                    job.setProject(project);
+                    job.setModelVersion(modelMapper.findByNameAndModelId(job.getModelVersionValue(), null));
+                    job.setOwner(userMapper.find(job.getOwnerId()));
+                }
+                return entities;
+            }
+        };
+        while (it.hasNext()) {
+            results.addAll(it.next());
+        }
+        return results;
+
+    }
+
+    public JobFlattenEntity findJobByUuid(String uuid) {
+        var andConditions = List.of(
+                (Object) TableQueryFilter.builder()
+                    .operator(TableQueryFilter.Operator.EQUAL)
+                    .operands(List.of(
+                        new TableQueryFilter.Column(IsDeletedColumn),
+                        new TableQueryFilter.Constant(ColumnTypeScalar.INT32, 0)))
+                    .build(),
+                TableQueryFilter.builder()
+                    .operator(TableQueryFilter.Operator.EQUAL)
+                    .operands(List.of(
+                        new TableQueryFilter.Column(KeyColumn),
+                        new TableQueryFilter.Constant(ColumnTypeScalar.STRING, uuid)))
+                    .build());
+        return this.findJobByCondition(andConditions);
+    }
+
+    public JobFlattenEntity findJobById(Long jobId) {
+        var andConditions = List.of(
+                (Object) TableQueryFilter.builder()
+                    .operator(TableQueryFilter.Operator.EQUAL)
+                    .operands(List.of(
+                        new TableQueryFilter.Column(IsDeletedColumn),
+                        new TableQueryFilter.Constant(ColumnTypeScalar.INT32, 0)))
+                    .build(),
+                TableQueryFilter.builder()
+                    .operator(TableQueryFilter.Operator.EQUAL)
+                    .operands(List.of(
+                        new TableQueryFilter.Column(LongIdColumn),
+                        new TableQueryFilter.Constant(ColumnTypeScalar.INT64, jobId)))
+                    .build());
+        return this.findJobByCondition(andConditions);
+    }
+
+    private JobFlattenEntity findJobByCondition(List<Object> conditions) {
```

Comment:
the method name should be `findUniqueJobByCondition` ? Or `List<JobFlattenEntity>` should be returned and the `"find multi eval jobs"` exception should not be there